general/solomiia_tymoshchuk/lecture2/_1/Calculator.java

```diff
@@ -0,0 +1,73 @@
+package general.solomiia_tymoshchuk.lecture2._1;
+
+import java.util.Scanner;
+
+/**
+ * Define a class that represents a calculator.
+ * The class should provide methods that accept another value and
+ * perform addition, subtraction, multiplication and division
+ * operations on a value stored in a calculator instance.
+ * Divides by a zero should be forbidden and ignored.
+ * The class should also provide a method to get a current value.
+ * The class should work with both integer and double numbers (ignore
+ * roundoff errors). Assume all operation results fit into the range of
+ * values for a current value type.
+ **/
+
+public class Calculator {
+    private double number;
+
+    public Calculator(double number){
+        this.number = number;
+    }
+
+    void validateNumber(double number1, String operator){
+        if(number1 == 0 && operator.equals("/")){
+            throw new ArithmeticException("Divided by zero operation cannot possible");
```

Comment:
throwing an exception is a costly operation, it would be better to inline this check into `calculateNumbersSwitchCase`:

```
if(number1 == 0 && operator.equals("/")){
    System.out.println("Divided by zero operation cannot possible");
    return;
}
```